sw/device/lib/testing/pinmux_testutils.c

```diff
@@ -0,0 +1,103 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/pinmux_testutils.h"
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/runtime/hart.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+#define PINMUX0_BASE_ADDR TOP_EARLGREY_PINMUX_AON_BASE_ADDR
+
+#define NUM_GPIO 32
+
+/**
+ * Checks that the given DIF call returns kDifOk. If the DIF call returns a
+ * different dif_result_t value (defined in sw/device/lib/dif/dif_base.h), this
+ * function aborts.
+ *
+ * A similar macro performs logging, but this is often not available during
+ * pinmux initialization, so this one does not.
+ *
+ * @param dif_call DIF call to invoke and check its return value.
+ * fails.
+ */
+#define CHECK_PINMUX_DIF_OK(dif_call)   \
+  do {                                  \
+    dif_result_t dif_result = dif_call; \
+    if (dif_result != kDifOk) {         \
+      abort();                          \
+    }                                   \
+  } while (false)
+
+void pinmux_init(void) {
+  dif_pinmux_t pinmux;
+  mmio_region_t pinmux_base_addr = mmio_region_from_addr(PINMUX0_BASE_ADDR);
+  CHECK_PINMUX_DIF_OK(dif_pinmux_init(pinmux_base_addr, &pinmux));
+  // TODO: This hardcoded configuration is temporary and needs to be
+  // replaced by proper initialization code
```

Comment:
Should we file an issue associated with this TODO?

Reply:
Added an explanation of what the function does and removed the TODO. Now it should be understood that this function provides a certain default config, but DIFs are available for finer-grained control.